### ISSUE TITLE
Unable to verify signature from event webhook

### ISSUE THREAD

**Issue by ASudre** [sendgrid/sendgrid-nodejs#1193]
### Issue Summary
I am trying to validate a webhook signed event. Unfortunately, it doesn't seem to work when SendGrid posts a sample JSON payload to my webhook listener.

Here is the request received on my webhook:
```bash
POST /webhooks/sendgrid HTTP/1.1
Host: c0aa5b77b698.ngrok.io
User-Agent: SendGrid Event API
Content-Length: 333
Content-Type: application/json;charset=utf-8
X-Twilio-Email-Event-Webhook-Signature: MEYCIQCxfnpzX3CftRSaaA4wHWHOKEyHbCf5jbwL/z/QQpncqQIhANu8Ug7FTuQhgGzhSVQvDSIS64rp+fXz2AloRnV5qcTL
X-Twilio-Email-Event-Webhook-Timestamp: 1599663888
Accept-Encoding: gzip
X-Forwarded-Proto: https
X-Forwarded-For: 167.89.119.83

[{"email":"arthur.sudre@stack-labs.com","event":"processed","send_at":0,"sg_event_id":"cHJvY2Vzc2VkLTE4NDg3OTEyLVhldDZiWjdXUlRHMmZZRGEtcDdhcnctMA","sg_message_id":"Xet6bZ7WRTG2fYDa-p7arw.filterdrecv-p3iad2-865cf6bb5-jhhw2-19-5F58E73D-D2.0","smtp-id":"<Xet6bZ7WRTG2fYDa-p7arw@ismtpd0007p1lon1.sendgrid.net>","timestamp":1599661886}]
```

My public key is: `MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEc1xJU2qhLFMxcOLdKWQIA2OZdmUTlNRT5xFEipLDnGkO0uhW8aJiIQxJGglBRiKWxNqm3jjRbUpaAaDH9WHkng==`

### Code Snippet
Given these information, I wrote the unit test:
```node
const { EventWebhook } = require('@sendgrid/eventwebhook');

function verify(publicKey, payload, signature, timestamp) {
  const ew = new EventWebhook();
  const key = ew.convertPublicKeyToECDSA(publicKey);
  return ew.verifySignature(key, payload, signature, timestamp);
}

describe('EventWebhook', () => {
  const PUBLIC_KEY = 'MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEc1xJU2qhLFMxcOLdKWQIA2OZdmUTlNRT5xFEipLDnGkO0uhW8aJiIQxJGglBRiKWxNqm3jjRbUpaAaDH9WHkng==';
  const SIGNATURE = 'MEYCIQCxfnpzX3CftRSaaA4wHWHOKEyHbCf5jbwL/z/QQpncqQIhANu8Ug7FTuQhgGzhSVQvDSIS64rp+fXz2AloRnV5qcTL';
  const TIMESTAMP = '1599663888';
  const PAYLOAD = JSON.stringify([
    {
      email: 'arthur.sudre@stack-labs.com',
      event: 'processed',
      send_at: 0,
      sg_event_id: 'cHJvY2Vzc2VkLTE4NDg3OTEyLVhldDZiWjdXUlRHMmZZRGEtcDdhcnctMA',
      sg_message_id: 'Xet6bZ7WRTG2fYDa-p7arw.filterdrecv-p3iad2-865cf6bb5-jhhw2-19-5F58E73D-D2.0',
      'smtp-id': '<Xet6bZ7WRTG2fYDa-p7arw@ismtpd0007p1lon1.sendgrid.net>',
      timestamp: 1599661886,
    },
  ]);

  describe('#verifySignature()', () => {
    it('should verify a valid signature', () => {
      expect(verify(
        PUBLIC_KEY,
        PAYLOAD,
        SIGNATURE,
        TIMESTAMP,
      )).toEqual(true);
    });
  });
});

```

### Exception/Log
The test fails. The verification is not valid.

### Technical details:
* "@sendgrid/eventwebhook": "^7.2.5",
* node version: v12.9.0

**Comment by childish-sambino** [comment]
Similar to https://github.com/sendgrid/sendgrid-csharp/issues/1018

tl;dr: Logging the request _hides_ the trailing `\r\n`. Try:
```
      expect(verify(
        PUBLIC_KEY,
        PAYLOAD + '\r\n',
        SIGNATURE,
        TIMESTAMP,
      )).toEqual(true);
```

We're looking into updating the examples to help with this.

**Comment by childish-sambino** [comment]
Fixed by https://github.com/sendgrid/sendgrid-nodejs/commit/e5c6fe4fddada920c198c776563438f4df7e6db9

**Comment by hakomori64** [comment]
I stuck into this problem and I found solution on this page
https://www.twilio.com/docs/runtime/quickstart/validate-webhook-requests-from-sendgrid

```
// Stringify the event and add newlines/carriage returns since they're expected by validator
  const rawEvent =
    JSON.stringify(sendGridPayload).split('},{').join('},\r\n{') + '\r\n';
```

**Comment by vikash-cision1** [comment]
i was using java library was stuck with signature validation. Below given 2 lines of code resolved payload issue. 
String payload = new ObjectMapper().writeValueAsString(bodyPayloadObject) + "\r\n";
payload = payload.replace("},{", "},\r\n{");
